### PR TITLE
feat(vm): Add resource `cloudavenue_vm_inserted_media` 

### PR DESCRIPTION
**Pull request by dmicheneau**
If you submit change in the provider code, please make sure to:

- [x] Write or modify examples in `examples/` directory
- [x] Write or modify acceptance tests
- [x] Run `make generate` to ensure the doc was updated properly

### How has this code been tested
```
TF_ACC=1 go test -v ./internal/tests/vm -run TestAccVMInsertedMediaResource
=== RUN   TestAccVMInsertedMediaResource
--- PASS: TestAccVMInsertedMediaResource (15.49s)
PASS
ok      github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/tests/vm  15.507s
```

### CODE REVIEW

File: internal/provider/vm/vm_inserted_media_resource.go [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+	myvapp "github.com/orange-cloudavenue/terraform-provider-cloudavenue/internal/provider/common/vapp"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource              = &vmInsertedMediaResource{}
+	_ resource.ResourceWithConfigure = &vmInsertedMediaResource{}
+)
+
+// NewVappVMInsertedMediaResource is a helper function to simplify the provider implementation.
```
Comment (azrod) [inline internal/provider/vm/vm_inserted_media_resource.go:26]: Bad comment

File: internal/provider/vm/vm_inserted_media_resource.go [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+	Name       types.String `tfsdk:"name"`
+	VappName   types.String `tfsdk:"vapp_name"`
+	VMName     types.String `tfsdk:"vm_name"`
+	EjectForce types.Bool   `tfsdk:"eject_force"`
+}
+
+// Metadata returns the resource type name.
+func (r *vmInsertedMediaResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vm_inserted_media"
+}
```
Comment (azrod) [inline internal/provider/vm/vm_inserted_media_resource.go:49]:
```suggestion
func (r *vmInsertedMediaResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
	resp.TypeName = req.ProviderTypeName + "_" + categoryName + "_" + "inserted_media"
}
```

File: internal/provider/vm/vm_inserted_media_resource.go [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+
+// Schema defines the schema for the resource.
+func (r *vmInsertedMediaResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The inserted_media resource allows you to manage a ...",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
```
Comment (azrod) [inline internal/provider/vm/vm_inserted_media_resource.go:56]: Please add Description

File: internal/provider/vm/vm_inserted_media_resource.go [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+
+// Metadata returns the resource type name.
+func (r *vmInsertedMediaResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vm_inserted_media"
+}
+
+// Schema defines the schema for the resource.
+func (r *vmInsertedMediaResource) Schema(ctx context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		MarkdownDescription: "The inserted_media resource allows you to manage a ...",
```
Comment (azrod) [inline internal/provider/vm/vm_inserted_media_resource.go:54]: Please finish this description

File: internal/provider/vm/vm_inserted_media_resource.go [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+			},
+			"vdc": schema.StringAttribute{
+				Optional:            true,
+				Computed:            true,
+				MarkdownDescription: "The name of VDC to use, optional if defined at provider level",
+			},
```
Comment (azrod) [inline internal/provider/vm/vm_inserted_media_resource.go:63]: VDC doesn't require replace ?

File: internal/provider/vm/vm_inserted_media_resource.go [hunk 6]
```diff
@@ -0,0 +1,431 @@
...
+	var isIsoMounted bool
+
+	for _, hardwareItem := range vm.VM.VirtualHardwareSection.Item {
+		if hardwareItem.ResourceType == int(15) { // 15 = CD/DVD Drive
+			isIsoMounted = true
+			break
+		}
+	}
+	if !isIsoMounted {
+		resp.Diagnostics.AddError("Media not mounted", "Media is not mounted on the VM")
```
Comment (azrod) [inline internal/provider/vm/vm_inserted_media_resource.go:267]: Add `resp.Diagnostics.RemoveResource(ctx)`

File: internal/tests/vm/vm_inserted_media_resource_test.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+			//	Check: resource.ComposeAggregateTestCheckFunc(
+			//		resource.TestCheckResourceAttrSet(resourceName, "id"),
+			//		resource.TestCheckResourceAttr(resourceName, "vdc", os.Getenv("CLOUDAVENUE_VDC")),
+			//		resource.TestCheckResourceAttr(resourceName, "catalog", "my-catalog_romain"),
+			//		resource.TestCheckResourceAttr(resourceName, "name", "debian-9.9.0-amd64-netinst.iso"),
+			//		resource.TestCheckResourceAttr(resourceName, "vapp_name", "vapp_test3"),
+			//		resource.TestCheckResourceAttr(resourceName, "vm_name", "TestRomain"),
+			//		resource.TestCheckResourceAttr(resourceName, "eject_force", "false"),
+			//	),
+			// },
```
Comment (azrod) [inline internal/tests/vm/vm_inserted_media_resource_test.go:58]: Remove your comment

File: internal/provider/vm/vm_inserted_media_resource.go [hunk 8]
```diff
@@ -0,0 +1,431 @@
+// Package vm provides a Terraform resource.
```
Comment (azrod) [inline internal/provider/vm/vm_inserted_media_resource.go:1]: Please rename your file `inserted_media_resource.go`

**Review by azrod — CHANGES_REQUESTED** [review]

File: docs/resources/vm_inserted_media.md [hunk 9]
```diff
@@ -0,0 +1,46 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "cloudavenue_vm_inserted_media Resource - cloudavenue"
+subcategory: ""
```
Comment (azrod) [inline docs/resources/vm_inserted_media.md:4]: Please add Subcategory

**Review by azrod — COMMENTED** [review]
.

File: internal/tests/vm/vm_inserted_media_resource_test.go [hunk 10]
```diff
@@ -0,0 +1,61 @@
+// Package vm provides the acceptance tests for the provider.
```
Comment (azrod) [inline internal/tests/vm/vm_inserted_media_resource_test.go:1]: Please rename your file `inserted_media_resource_test.go`

**Review by azrod — COMMENTED** [review]
.

File: internal/provider/vm/vm_inserted_media_resource.go (hunk 6 above)
Comment (dmicheneau) [reply internal/provider/vm/vm_inserted_media_resource.go:267]: fix by: resp.State.RemoveResource(ctx)

File: internal/client/client.go [hunk 11]
```diff
@@ -63,7 +63,7 @@ func (c *CloudAvenue) New() (*CloudAvenue, error) {
 		return nil, fmt.Errorf("%w : %v", ErrConfigureVmware, err)
 	}
 
-	c.Vmware = govcd.NewVCDClient(*c.urlVmware, false)
+	c.Vmware = govcd.NewVCDClient(*c.urlVmware, false, govcd.WithAPIVersion("37.0"))
```
Comment (azrod) [inline internal/client/client.go:66]: Please create specific PR for this change

File: templates/resources/vm_inserted_media.md.tmpl [hunk 12]
```diff
@@ -0,0 +1,24 @@
+---
+page_title: "{{.Name}} {{.Type}} - {{.ProviderName}}"
+subcategory: "VM"
+description: |-
+{{ .Description | plainmarkdown | trimspace | prefixlines "  " }}
+
+  Only one media is allowed per VM.
```
Comment (azrod) [inline templates/resources/vm_inserted_media.md.tmpl:7]:
```suggestion
~> Note:  Only one media is allowed per VM.Only one media is allowed per VM.
```

File: templates/resources/vm_inserted_media.md.tmpl [hunk 13]
```diff
@@ -0,0 +1,24 @@
+---
+page_title: "{{.Name}} {{.Type}} - {{.ProviderName}}"
+subcategory: "VM"
```
Comment (azrod) [inline templates/resources/vm_inserted_media.md.tmpl:3]:
```suggestion
subcategory: "VM (Virtual Machine)"
```

**Review by azrod — CHANGES_REQUESTED** [review]

File: templates/resources/vm_inserted_media.md.tmpl [hunk 14]
```diff
@@ -0,0 +1,24 @@
+---
+page_title: "{{.Name}} {{.Type}} - {{.ProviderName}}"
+subcategory: "VM (Virtual Machine)"
+description: |-
+{{ .Description | plainmarkdown | trimspace | prefixlines "  " }}
+
+  Note: Only one media is allowed per VM.
```
Comment (azrod) [inline templates/resources/vm_inserted_media.md.tmpl:7]: `~>` Add Warning Block in terraform doc. 
Preview result in [doc-preview](https://registry.terraform.io/tools/doc-preview)
```suggestion
 ~> Note: Only one media is allowed per VM.
```

**Review by azrod — CHANGES_REQUESTED** [review]

File: templates/resources/vm_inserted_media.md.tmpl [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+description: |-
+{{ .Description | plainmarkdown | trimspace | prefixlines "  " }}
+
+---
+
+# {{.Name}} ({{.Type}})
+
+{{ .Description | trimspace }}
+
+  ~> Note: Only one media is allowed per VM.
```
Comment (azrod) [inline templates/resources/vm_inserted_media.md.tmpl:13]:
```suggestion
~> Only one media is allowed per VM.
```

**Review by azrod — APPROVED** [review]

**Pull request merged**